config/confighttp/confighttp.go

```diff
@@ -187,7 +190,7 @@ func WithErrorHandler(e middleware.ErrorHandler) ToServerOption {
 }
 
 // ToServer creates an http.Server from settings object.
-func (hss *HTTPServerSettings) ToServer(handler http.Handler, settings component.TelemetrySettings, opts ...ToServerOption) *http.Server {
+func (hss *HTTPServerSettings) ToServer(host component.Host, handler http.Handler, settings component.TelemetrySettings, opts ...ToServerOption) (*http.Server, error) {
```

Comment:
Can we fix the order between host/handler/settings? I think in grpc we use host, settings, other things. Lets keep that here as well?